components/external-services-mock/internal/notification/handler.go

```diff
@@ -116,56 +116,85 @@ func NewHandler(notificationConfiguration NotificationsConfiguration) *Handler {
-		return
-	}
+// NewPatchHandler returns a handler function that handles synchronous formation assignment notification requests for Assign operation.
+// If shouldHaveStateInBody is 'true' the handler will add State in the response body and if it is 'false' it won't.
+func (h *Handler) NewPatchHandler(shouldHaveStateInBody bool) func(http.ResponseWriter, *http.Request) {
+	return func(writer http.ResponseWriter, r *http.Request) {
+		id, ok := mux.Vars(r)["tenantId"]
+		if !ok {
+			httphelpers.WriteError(writer, errors.New("missing tenantId in url"), http.StatusBadRequest)
+			return
+		}
 
-	if _, ok = h.Mappings[id]; !ok {
-		h.Mappings[id] = make([]Response, 0, 1)
-	}
-	bodyBytes, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		httphelpers.WriteError(writer, errors.Wrap(err, "error while reading request body"), http.StatusInternalServerError)
-		return
-	}
+		if _, ok = h.Mappings[id]; !ok {
+			h.Mappings[id] = make([]Response, 0, 1)
+		}
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			httphelpers.WriteError(writer, errors.Wrap(err, "error while reading request body"), http.StatusInternalServerError)
+			return
+		}
 
-	var result interface{}
-	if err := json.Unmarshal(bodyBytes, &result); err != nil {
-		httphelpers.WriteError(writer, errors.Wrap(err, "body is not a valid JSON"), http.StatusBadRequest)
-		return
-	}
-	mappings := h.Mappings[id]
-	mappings = append(h.Mappings[id], Response{
-		Operation:   Assign,
-		RequestBody: bodyBytes,
-	})
-	h.Mappings[id] = mappings
+		var result interface{}
+		if err := json.Unmarshal(bodyBytes, &result); err != nil {
+			httphelpers.WriteError(writer, errors.Wrap(err, "body is not a valid JSON"), http.StatusBadRequest)
+			return
+		}
+		mappings := h.Mappings[id]
+		mappings = append(h.Mappings[id], Response{
+			Operation:   Assign,
+			RequestBody: bodyBytes,
+		})
+		h.Mappings[id] = mappings
 
-	response := struct {
-		Config struct {
-			Key  string `json:"key"`
-			Key2 struct {
-				Key string `json:"key"`
-			} `json:"key2"`
+		var response interface{}
+		if !shouldHaveStateInBody {
+			response = struct {
+				Config struct {
+					Key  string `json:"key"`
+					Key2 struct {
+						Key string `json:"key"`
+					} `json:"key2"`
+				}
+			}{
+				Config: struct {
+					Key  string `json:"key"`
+					Key2 struct {
+						Key string `json:"key"`
+					} `json:"key2"`
+				}{
+					Key: "value",
+					Key2: struct {
+						Key string `json:"key"`
+					}{Key: "value2"},
+				},
+			}
+		} else {
+			response = struct {
+				State  string `json:"state"`
+				Config struct {
+					Key  string `json:"key"`
+					Key2 struct {
+						Key string `json:"key"`
+					} `json:"key2"`
+				}
+			}{
+				State: "CONFIG_PENDING",
+				Config: struct {
+					Key  string `json:"key"`
+					Key2 struct {
+						Key string `json:"key"`
+					} `json:"key2"`
+				}{
+					Key: "value",
+					Key2: struct {
+						Key string `json:"key"`
+					}{Key: "value2"},
+				},
+			}
```

Comment:
Isn't there a better approach instead of this `if`-`else` approach?(Valid for the other handlers as well). For example - to use a separate handler instead of modifying the existing one with this `shouldHaveStateInBody` property. Similar to the async handlers we could extract the main logic into a function and define a "response function" that in the "with-state" handler will be configured to send `state` but in the "normal handler"(patch/delete) will send only status code?

Also, could we define a typed stucture(similar to the ones for the request body - `FormationAssignmentRequestBody` and `FormationRequestBody`) instead of using an anonymous one? Even before when we had only one response it was hard/ugly to read it. But now we duplicate the same thing with an additional property.